front-end-tools/src/components/WriteComponent.tsx

```diff
@@ -0,0 +1,618 @@
+            if (entryPointName === undefined) {
+                throw new Error('Set entry point name');
+            }
+
+            if (smartContractName === undefined) {
+                throw new Error('Set smart contract name');
+            }
+
+            let schema = '';
+
+            const schemaFromModule = deriveContractInfo ? embeddedModuleSchemaBase64 : uploadedModuleSchemaBase64;
+
+            if (schemaFromModule !== undefined) {
+                schema = schemaFromModule;
+            }
+
+            const functionTemplate = getUpdateContractParameterSchema(
+                toBuffer(schema, 'base64'),
+                smartContractName,
+                entryPointName
+            );
+
+            receiveTemplate = displayTypeSchemaTemplate(functionTemplate);
+
+            setEntryPointTemplate(receiveTemplate);
+        } catch (e) {
+            if (deriveContractInfo) {
+                setSchemaError(
+                    `Could not derive the embedded schema from the smart contract index. Uncheck "Derive From Smart Contract Index" checkbox to manually upload a schema or uncheck "Has Input Paramter" checkbox if this entrypoint has no input parameter. Original error: ${e}`
+                );
+            } else {
+                setSchemaError(
+                    `Could not get schema from uploaded schema. Uncheck "Has Input Paramter" checkbox if this entrypoint has no input parameter. Original error: ${e}`
+                );
+            }
+        }
+
+        if (receiveTemplate) {
+            if (inputParameterType === 'array') {
+                form.setValue('inputParameter', JSON.stringify(JSON.parse(receiveTemplate), undefined, 2));
+            } else if (inputParameterType === 'object') {
+                form.setValue('inputParameter', JSON.stringify(JSON.parse(receiveTemplate), undefined, 2));
+            }
+        }
+    }, [entryPointName, hasInputParameter, smartContractName, uploadedModuleSchemaBase64, inputParameterType]);
+
+    function onSubmit(data: FormType) {
+        setTxHashUpdate(undefined);
+        setTransactionErrorUpdate(undefined);
+        setTransactionOutcome(undefined);
+
+        const schema = deriveContractInfo ? embeddedModuleSchemaBase64 : uploadedModuleSchemaBase64;
+
+        // Send update transaction
+
+        const tx = write(
+            connection,
+            account,
+            data.inputParameter,
+            data.smartContractName,
+            data.entryPointName,
+            data.hasInputParameter,
+            data.deriveFromSmartContractIndex,
+            schema,
+            data.inputParameterType,
+            BigInt(data.maxExecutionEnergy),
+            BigInt(data.smartContractIndex),
+            data.cCDAmount ? BigInt(data.cCDAmount) : BigInt(0)
+        );
+
+        tx.then(setTxHashUpdate).catch((err: Error) => setTransactionErrorUpdate((err as Error).message));
+    }
+
+    return (
+        <Box header="Write To Smart Contract">
```

Comment:
It would be better to call this `update smart contract instance`. That would fit with the terminology we use elsewhere.